python/tank/bootstrap/baked_configuration.py

```diff
@@ -0,0 +1,136 @@
+from .. import constants
+import cPickle as pickle
+
+from ..util import ShotgunPath
+
+log = LogManager.get_logger(__name__)
+
+class BakedConfiguration(Configuration):
+    """
+    Represents a configuration that has been baked out at build time,
+    containing incomplete state data - at bake time we don't yet know
+    what project id, what site, configuration id etc. will be required.
+    """
+
+    def __init__(
+            self,
+            path,
+            sg,
+            project_id,
+            entry_point,
+            pipeline_config_id,
+            bundle_cache_fallback_paths
+    ):
+        """
+        Constructor.
+
+        :param path: ShotgunPath object describing the path to this configuration
+        :param sg: Shotgun API instance
+        :param project_id: Project id for the shotgun project associated with the
+                           configuration. For a site-level configuration, this
+                           can be set to None.
+        :param entry_point: Entry point string to identify the scope for a particular plugin
+                            or integration. For more information,
+                            see :meth:`~sgtk.bootstrap.ToolkitManager.entry_point`. For
+                            non-plugin based toolkit projects, this value is None.
+        :param pipeline_config_id: Pipeline Configuration id for the shotgun
+                                   pipeline config id associated. If a config does
+                                   not have an associated entity in Shotgun, this
+                                   should be set to None.
+        :param bundle_cache_fallback_paths: List of additional paths where apps are cached.
+        """
+        super(BakedConfiguration, self).__init__(path)
+        self._path = path
+        self._sg_connection = sg
+        self._project_id = project_id
+        self._entry_point = entry_point
+        self._pipeline_config_id = pipeline_config_id
+        self._bundle_cache_fallback_paths = bundle_cache_fallback_paths
+
+    def __str__(self):
+        """
+        User friendly representation of the config
+        """
+        return "Pre-baked config '%s'" % self._path
+
+    def __repr__(self):
+        """
+        Low level representation of the config.
+        """
+        return "<Pre-baked config '%s'>" % self._path
+
+    def status(self):
+        """
+        Compares the actual configuration installed on disk against the
+        associated configuration described by the descriptor passed in via
+        the class constructor.
+
+        :returns: LOCAL_CFG_UP_TO_DATE, LOCAL_CFG_MISSING,
+                  LOCAL_CFG_DIFFERENT, or LOCAL_CFG_INVALID
+        """
+        return self.LOCAL_CFG_UP_TO_DATE
+
+    def update_configuration(self):
+        """
+        Ensure that the configuration is up to date with the one
+        given by the associated descriptor.
+
+        This method fails gracefully and attempts to roll back to a
```

Comment:
This sentence should probably be removed. Perhaps update this docstring entirely noting that for baked configs, this is a no-op.